File: shared/contracts/transcript_report.py
"""Shared transcript report contract."""

from __future__ import annotations

from dataclasses import dataclass, asdict, field
from typing import Any


@dataclass(slots=True)
class TranscriptReport:
    """Describe a transcript intelligence report and its derived insights."""

    report_id: str
    archive_name: str
    created_at: float
    conversation_count: int
    languages: list[str] = field(default_factory=list)
    missing_intents: list[dict[str, Any]] = field(default_factory=list)
    procedure_summaries: list[dict[str, Any]] = field(default_factory=list)
    faq_entries: list[dict[str, Any]] = field(default_factory=list)
    workflow_suggestions: list[dict[str, Any]] = field(default_factory=list)
    suggested_tests: list[dict[str, Any]] = field(default_factory=list)
    insights: list[dict[str, Any]] = field(default_factory=list)
    knowledge_asset: dict[str, Any] = field(default_factory=dict)
    conversations: list[dict[str, Any]] = field(default_factory=list)
    intent_accuracy: float | None = None
    intent_accuracy_samples: int = 0

    def to_dict(self) -> dict[str, Any]:
        """Return a JSON-safe representation for API and CLI persistence."""
        payload = asdict(self)
        if not payload["conversation_count"]:
            payload["conversation_count"] = len(self.conversations)
        return payload

    @classmethod
    def from_dict(cls, data: dict[str, Any]) -> TranscriptReport:
        """Rehydrate a transcript report from persisted data."""
        conversations = list(data.get("conversations", []))
        return cls(
            report_id=data["report_id"],
            archive_name=data["archive_name"],
            created_at=float(data["created_at"]),
            conversation_count=int(data.get("conversation_count", len(conversations))),
            languages=list(data.get("languages", [])),
            missing_intents=list(data.get("missing_intents", [])),
            procedure_summaries=list(data.get("procedure_summaries", [])),
            faq_entries=list(data.get("faq_entries", [])),
            workflow_suggestions=list(data.get("workflow_suggestions", [])),
            suggested_tests=list(data.get("suggested_tests", [])),
            insights=list(data.get("insights", [])),
            knowledge_asset=dict(data.get("knowledge_asset", {})),
            conversations=conversations,
            intent_accuracy=data.get("intent_accuracy"),
            intent_accuracy_samples=int(data.get("intent_accuracy_samples", 0)),
        )

File: shared/contracts/transcript-report.ts
export interface TranscriptConversation {
  conversation_id: string;
  session_id: string;
  user_message: string;
  agent_response: string;
  outcome: string;
  language: string;
  intent: string;
  transfer_reason: string | null;
  source_file: string;
  procedure_steps: string[];
}

export interface MissingIntent {
  intent: string;
  count: number;
  reason: string;
}

export interface TranscriptInsight {
  insight_id: string;
  title: string;
  summary: string;
  recommendation: string;
  drafted_change_prompt: string;
  metric_name: string;
  share: number;
  count: number;
  total: number;
  evidence: string[];
}

export interface TranscriptProcedureSummary {
  intent: string;
  steps: string[];
  source_conversation_id: string;
}

export interface TranscriptFaqEntry {
  intent: string;
  question: string;
  answer: string;
}

export interface TranscriptWorkflowSuggestion {
  title: string;
  description: string;
}

export interface TranscriptSuggestedTest {
  name: string;
  user_message: string;
  expected_behavior: string;
}

export interface TranscriptKnowledgeAssetSummary {
  asset_id: string;
  entry_count: number;
}

export interface TranscriptReport {
  report_id: string;
  archive_name: string;
  created_at: string;
  conversation_count: number;
  languages: string[];
  missing_intents: MissingIntent[];
  procedure_summaries: TranscriptProcedureSummary[];
  faq_entries: TranscriptFaqEntry[];
  workflow_suggestions: TranscriptWorkflowSuggestion[];
  suggested_tests: TranscriptSuggestedTest[];
  insights: TranscriptInsight[];
  knowledge_asset: TranscriptKnowledgeAssetSummary;
  conversations: TranscriptConversation[];
  intent_accuracy?: number | null;
  intent_accuracy_samples?: number;
  source?: 'cli' | 'api' | 'ui';
  metadata?: Record<string, unknown>;
}

export interface TranscriptReportSummary {
  report_id: string;
  archive_name: string;
  created_at: string;
  conversation_count: number;
  languages: string[];
  knowledge_asset?: TranscriptKnowledgeAssetSummary;
}
